### PR TITLE
test(rust/esbuild/default): enable re_export_default_external.

### PR DESCRIPTION
**Pull request by Cecil0o0**

### CODE REVIEW

File: crates/rolldown/tests/esbuild/default/re_export_default_external_common_js/artifacts.snap [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+## entry_js.cjs
+
+```js
+const { __toESM } = require("./$runtime$.cjs");
+const { default: foo } = __toESM(require("foo"));
+const { default: bar } = __toESM(require("bar"));
+
+exports.bar = bar;
+exports.foo = foo;
+```
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/default/re_export_default_external_common_js/artifacts.snap:17]: Please recover this test case, 
the esbuild snapshot is
```js
================================================================================
TestReExportDefaultExternalCommonJS
---------- /out.js ----------
// entry.js
var entry_exports = {};
__export(entry_exports, {
  bar: () => import_bar.default,
  foo: () => import_foo.default
});
module.exports = __toCommonJS(entry_exports);
var import_foo = __toESM(require("foo"));

// bar.js
var import_bar = __toESM(require("bar"));

```

File: crates/rolldown/tests/esbuild/default/re_export_default_external_es6/artifacts.snap [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# Assets
+
+## entry_js.mjs
+
+```js
+import { default as foo } from "foo";
+import { default as bar } from "bar";
+
+export { bar, foo };
+```
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/default/re_export_default_external_es6/artifacts.snap:15]: Could you solve the `unresolved` error? esbuild snapshot does not have these errors.

Comment (Cecil0o0) [reply crates/rolldown/tests/esbuild/default/re_export_default_external_es6/artifacts.snap:15]: Ok, will be resolved tomorrow

File: crates/rolldown/tests/esbuild/default/re_export_default_external_common_js/artifacts.snap (hunk 1 above)
Comment (Cecil0o0) [reply crates/rolldown/tests/esbuild/default/re_export_default_external_common_js/artifacts.snap:17]: could I ask, "recover this test case" means not to enable this case until snapshot be same with esbuild's?

Comment (hyf0) [reply crates/rolldown/tests/esbuild/default/re_export_default_external_common_js/artifacts.snap:17]: No need to revert. I will take a note on this case and fix it later. It's fine we merge them first. But still could you solve the `UNRESOLVED` warning.

Comment (Cecil0o0) [reply crates/rolldown/tests/esbuild/default/re_export_default_external_common_js/artifacts.snap:17]: Got it, I solved them.

File: crates/rolldown/tests/esbuild/default/re_export_default_external_common_js/_config.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
+  "expectExecuted": false,
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/default/re_export_default_external_common_js/_config.json:2]: Why do you mark this test case as none executed?

**Review by hyf0 — APPROVED** [review]
Thanks

**Pull request merged**